### PR TITLE
Feature/OP-859

### PR DESCRIPTION
**Pull request by joelbcastillo**
Permission for OpenRecords

### CODE REVIEW

File: app/response/views.py [hunk 1]
```diff
@@ -180,14 +195,17 @@ def response_closing(request_id):
             flash('Uh Oh, it looks like the closing {} is missing! '
                   'This is probably NOT your fault.'.format(field), category='danger')
             return redirect(url_for('request.view', request_id=request_id))
-
-        add_closing(request_id,
+        try:
+            add_closing(request_id,
                     flask_request.form.getlist('reasons'),
```
Comment (Palisand) [inline app/response/views.py]: align this please

File: app/templates/response/add_closing.js.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
             reason_ids.focus();
             $(this).prop('selected', !$(this).prop('selected'));
             return false;
-        });
+        });
```
Comment (Palisand) [inline app/templates/response/add_closing.js.html]: don't know how this made it through, pycharm's usually pretty good  with this type of thing...

File: app/user_request/utils.py [hunk 3]
```diff
@@ -60,3 +186,17 @@ def remove_user_request(request_id, user_guid):
...
+def _get_agency_admin_emails(request_id):
+    """
+    Retrieve a list of agency administrator emails
+    :param request_id: FOIL request id
+    :return: list(Agency Adminstrator Emails)
+    """
+
+    agency_ein = Requests.query.filter_by(id=request_id).one().agency.ein
+
+    agency_administrators = Users.query.filter_by(agency_ein=agency_ein, is_agency_admin=True).all()
```
Comment (Palisand) [inline app/user_request/utils.py]: `is_agency_active=True` as well?

File: app/lib/jinja_filters.py [hunk 4]
```diff
@@ -1,6 +1,5 @@
 from app.constants import response_type, response_privacy
```
Comment (joelbcastillo) [inline app/lib/jinja_filters.py]: 👍

File: app/models.py [hunk 5]
```diff
@@ -51,53 +53,66 @@ def populate(cls):
...
             role_name.ANONYMOUS: (
-                permission.DUPLICATE_REQUEST |
-                permission.VIEW_REQUEST_STATUS_PUBLIC |
-                permission.VIEW_REQUEST_INFO_PUBLIC
-            ),
-            role_name.PUBLIC_NON_REQUESTER: (
-                permission.DUPLICATE_REQUEST |
-                permission.VIEW_REQUEST_STATUS_PUBLIC |
-                permission.VIEW_REQUEST_INFO_PUBLIC
+                0
```
Comment (joelbcastillo) [inline app/models.py]: 👍

File: app/permissions/views.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+from app.permissions.utils import has_permission
+
+
+@permissions.route('/<role>', methods=['GET'])
+def get_role_permissions(role):
+    """
+    Given a role, return the permissions that are in that role.
+
+    :param role:
+    :return:
```
Comment (joelbcastillo) [inline app/permissions/views.py]: 👍

File: app/response/views.py (hunk 1 above)
Comment (joelbcastillo) [inline app/response/views.py]: 👍

File: app/templates/response/add_closing.js.html (hunk 2 above)
Comment (joelbcastillo) [inline app/templates/response/add_closing.js.html]: 👍

**Pull request merged**